lxd/storage/drivers/driver_ceph_volumes.go

```diff
@@ -0,0 +1,1028 @@
+
+	return nil
+}
+
+func (d *ceph) RefreshVolume(vol Volume, srcVol Volume, srcSnapshots []Volume, op *operations.Operation) error {
+	return genericCopyVolume(d, nil, vol, srcVol, srcSnapshots, true, op)
+}
+
+func (d *ceph) DeleteVolume(vol Volume, op *operations.Operation) error {
+	if vol.volType == VolumeTypeImage {
+		// Try to umount but don't fail
+		d.UnmountVolume(vol, op)
+
+		// Check if image has dependant snapshots
+		_, err := d.rbdListSnapshotClones(vol, "readonly")
+		if err != nil {
+			if err != db.ErrNoSuchObject {
+				return err
+			}
+
+			// Unprotect snapshot
+			err = d.rbdUnprotectVolumeSnapshot(vol, "readonly")
+			if err != nil {
+				return err
+			}
+
+			// Delete snapshots
+			_, err = shared.RunCommand(
+				"rbd",
+				"--id", d.config["ceph.user.name"],
+				"--cluster", d.config["ceph.cluster_name"],
+				"--pool", d.config["ceph.osd.pool_name"],
+				"snap",
+				"purge",
+				d.getRBDVolumeName(vol, "", false, false))
+			if err != nil {
+				return err
+			}
+
+			// Unmap image
+			err = d.rbdUnmapVolume(vol, true)
+			if err != nil {
+				return err
+			}
+
+			// Delete image
+			err = d.rbdDeleteVolume(vol)
+		} else {
+			err = d.rbdUnmapVolume(vol, true)
+			if err != nil {
+				return err
+			}
+
+			err = d.rbdMarkVolumeDeleted(vol, vol.name, vol.config["block.filesystem"])
+		}
+		if err != nil {
+			return err
+		}
+	} else {
+		if !d.HasVolume(vol) {
+			return nil
+		}
+
+		_, err := d.UnmountVolume(vol, op)
+		if err != nil {
+			return err
+		}
+
+		ret := d.deleteVolume(vol)
+		if ret < 0 {
+			return fmt.Errorf("Failed to delete volume")
+		}
+	}
+
+	err := wipeDirectory(vol.MountPath())
+	if err != nil {
+		return err
+	}
+
+	err = os.Remove(vol.MountPath())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *ceph) RenameVolume(vol Volume, newName string, op *operations.Operation) error {
+	revert := revert.New()
+	defer revert.Fail()
+
+	_, err := d.UnmountVolume(vol, op)
+	if err != nil {
+		return err
+	}
+
+	err = d.rbdUnmapVolume(vol, true)
+	if err != nil {
+		return nil
+	}
+
+	revert.Add(func() { d.rbdMapVolume(vol) })
+
+	err = d.rbdRenameVolume(vol, newName)
+	if err != nil {
+		return err
+	}
+
+	newVol := NewVolume(d, d.name, vol.volType, vol.contentType, newName, nil, nil)
+
+	revert.Add(func() { d.rbdRenameVolume(newVol, vol.name) })
+
+	_, err = d.rbdMapVolume(newVol)
+	if err != nil {
+		return err
+	}
+
+	err = genericVFSRenameVolume(d, vol, newName, op)
+	if err != nil {
+		return nil
+	}
+
+	revert.Success()
+	return nil
+}
+
+func (d *ceph) UpdateVolume(vol Volume, changedConfig map[string]string) error {
+	if vol.volType != VolumeTypeCustom {
+		return ErrNotSupported
+	}
+
+	val, ok := changedConfig["size"]
+	if ok {
+		err := d.SetVolumeQuota(vol, val, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *ceph) GetVolumeUsage(vol Volume) (int64, error) {
+	return -1, fmt.Errorf("RBD quotas are currently not supported")
```

Comment:
@monstermunchkin there is a constant `ErrNotSupported` for this. This is then recognised by the caller and doesn't spam the syslog with errors :)